### PR TITLE
e2e test for sas token

### PR DESCRIPTION
**Pull request by cvvz**
**What type of PR is this?**

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:
- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:
```
none
```

### CODE REVIEW

File: pkg/blob/blob.go [hunk 1]
```diff
@@ -317,7 +317,7 @@ func checkContainerNameBeginAndEnd(containerName string) bool {
 
 // isSASToken checks if the key contains the patterns. Because a SAS Token must have these strings, use them to judge.
 func isSASToken(key string) bool {
-	return strings.Contains(key, "?sv=")
+	return strings.HasPrefix(key, "?")
```
Comment (andyzhangx) [inline pkg/blob/blob.go:320]: is the previous test failure due to this issue?

Comment (andyzhangx) [inline pkg/blob/blob.go:320]: add a link to sas token format: https://docs.microsoft.com/en-us/rest/api/eventhub/generate-sas-token
I think it's better set like this: `strings.Contains(key, "&sig=")` per the above sas token format doc

Comment (cvvz) [reply pkg/blob/blob.go:320]: It's one of the reason. The other one is my mistake.

Comment (cvvz) [reply pkg/blob/blob.go:320]: The SAS token is actually a query string in URL, the format is like: `?sv={}&ss={}&srt={}&sp={}&se={}&st={}&spr={}&sig={}`. I guess maybe `sig=` could be put right after question mark `?`. Also I think question mark won't appear in access token, so I make the change in this way.

Comment (cvvz) [reply pkg/blob/blob.go:320]: Also I will add a link to SAS token format in the comment.

Comment (andyzhangx) [reply pkg/blob/blob.go:321]: the real sas token format actually only contains a few of the above const format, we may only check a few, e.g. `se=`, `sig=`,  and question mark check may be not necessary, as long as the former 2 fixed strings are there, then it's a sas token

Comment (andyzhangx) [reply pkg/blob/blob.go:321]: nvm, let's just use prefix `?` check to make it simple.

File: test/e2e/testsuites/pre_provisioned_sastoken_tester.go [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+	sasURL, err := serviceClient.GetSASURL(
+		azblob.AccountSASResourceTypes{Object: true, Service: true, Container: true},
+		azblob.AccountSASPermissions{Read: true, List: true, Write: true, Delete: true, Add: true, Create: true, Update: true},
+		time.Now(), time.Now().Add(10*time.Hour))
+	framework.ExpectNoError(err)
+	ginkgo.By("SAS URL: " + sasURL)
+	u, err := url.Parse(sasURL)
+	framework.ExpectNoError(err)
+	sasToken := "?" + u.RawQuery
+	ginkgo.By("SAS Token: " + sasToken)
```
Comment (andyzhangx) [inline test/e2e/testsuites/pre_provisioned_sastoken_tester.go:105]: remove any secret and sas token print log

**Review by andyzhangx — APPROVED** [review]
/lgtm
/approve

**Pull request merged**